Add custom hit-shapes & TargetableOffsets to TS structures

reaperrr opened the pull request.

reaperrr committed:
Add TS default hit-shapes

reaperrr committed:
Add TS civilian hit-shapes

reaperrr committed:
Add TS base-building hit-shapes

pchote commented on mods/ts/rules/civilian-structures.yaml:
```diff
@@ -18,6 +19,18 @@ ABAN01:
 
 ABAN02:
 	Inherits: ^CivBuilding
+	HitShape:
```
Are you sure this is correct?

![screen shot 2017-06-06 at 22 20 10](https://user-images.githubusercontent.com/167819/26852551-8fee4ac0-4b06-11e7-8176-4541f4ce6119.png)

reaperrr replied:
Eh, no. Forgot to finish this one before pushing, although not using a single 5x3 rectangle is intentional (I wanted to match the art).

reaperrr committed:
Tweak sprite offsets of some TS aban civ buildings

In cases where the sprite has enough margin to the footprint cell edges, moving down the sprite a bit is much easier than manually setting up custom targetable offsets.

pchote commented:
We won't be shipping TS with the playtest, so moving to the Next Release milestone - we need to fix this in the short term, but it isn't a blocker.

rob-v commented on mods/ts/rules/defaults.yaml:
```diff
@@ -123,6 +124,114 @@
 	ExternalCondition@CRATE-CLOAK:
 		Condition: crate-cloak
 
+^1x2Shape:
```
2x1 as in TD?

reaperrr replied:
I wanted to match the Building.Footprint, but I guess it would be more intuitive to change the LocalYaw rotation into the other direction and switch these around like you suggested.

rob-v commented on mods/ts/rules/defaults.yaml:
```diff
@@ -123,6 +124,114 @@
+			TopLeft: -1536, -512
+			BottomRight: 1536, 512
+
+^2x3Shape:
```
3x2 as in TD? etc for other shapes

rob-v commented on mods/ts/rules/nod-support.yaml:
```diff
@@ -270,6 +271,19 @@ NASAM:
+		Type: Rectangle
+			TopLeft: 0, -1536
+			BottomRight: 1024, 1536
+	HitShape@lowerSphere:
```
I need to find out where/how are these additional 'Hitshape@name' used, style nit: LowerSphere?

rob-v reviewed (commented):
It would be better to unify default shapes with TD re width x heigth.

GraionDilach commented on mods/ts/rules/nod-support.yaml:
```diff
@@ -270,6 +271,19 @@ NASAM:
 	Inherits: ^Building
 	Inherits@IDISABLED: ^DisabledOverlay
+	HitShape:
+		UseOccupiedCellsOffsets: false
```
Is there a particular reason why every time I see a composite setup in the PRs, this ends up disabled? I couldn't notice anything wrong if I used additional hitboxes without targetable offsets.

reaperrr replied:
If the main shape is smaller than the occupied area, some targetable positions would otherwise be located outside (or at least too close to the border of) the shape.

reaperrr commented:
Rebased and updated.
Still depends on #13553, but otherwise this PR is ready for review now.

reaperrr commented:
Updated and rebased.

abcdefg30 commented on mods/ts/rules/civilian-structures.yaml:
```diff
@@ -380,10 +403,11 @@ BBOARD16:
 		Name: Rade's Roadhouse
 	Building:
-		Footprint: xxx xxx xxx
+		Footprint: xxu xxX XXX
```
There's still a `u` in here.

reaperrr commented:
Fixed.

abcdefg30 reviewed (dismissed):
![image](https://user-images.githubusercontent.com/7704140/28241066-8d07aab8-698d-11e7-87d7-d33bdc3a7bd1.png)

abcdefg30 commented:
Bonus points if you also update the old base buildings. :)

reaperrr commented:
Updated.

pchote commented:
I made a test map ([dropbox link](https://www.dropbox.com/s/noo8jx6ee9jk38j/actors.oramap?dl=0)) and spotted a few last issues:
![screen shot 2017-07-23 at 11 15 36](https://user-images.githubusercontent.com/167819/28498589-c54f4d20-6f98-11e7-8cb7-0a892e94d024.png)
![screen shot 2017-07-23 at 11 15 41](https://user-images.githubusercontent.com/167819/28498588-c54cf71e-6f98-11e7-9c82-714003d1d0e9.png)
![screen shot 2017-07-23 at 11 16 13](https://user-images.githubusercontent.com/167819/28498594-cf12df16-6f98-11e7-891f-59842720ec6d.png)
![screen shot 2017-07-23 at 11 16 53](https://user-images.githubusercontent.com/167819/28498593-cf109c4c-6f98-11e7-823b-19d702744167.png)

reaperrr committed:
Center two-legged billboard sprites

They are supposed to be positioned in the middle of the two cells they occupy.

reaperrr commented:
Updated, fixed all of the above plus a few last ones I noticed myself. This should be ready to merge now.

![bboard1](https://user-images.githubusercontent.com/2857877/28748456-f49b4820-74b8-11e7-9682-240546eeac6a.png)
![bboard2](https://user-images.githubusercontent.com/2857877/28748457-fb7486e8-74b8-11e7-933b-81fa9b740332.png)
![solar1](https://user-images.githubusercontent.com/2857877/28748458-fe51f56c-74b8-11e7-9a47-e483f661ca45.png)
![solar2](https://user-images.githubusercontent.com/2857877/28748460-ffdcb642-74b8-11e7-98db-81ee7e3e4c44.png)
![kodiak](https://user-images.githubusercontent.com/2857877/28748461-019f4968-74b9-11e7-96c7-b0e46a35b314.png)

ltem commented:
![image](https://user-images.githubusercontent.com/20945684/28756653-e0a028bc-7572-11e7-8d7e-eda18e4fc155.png)

ltem commented:
Some nits but despite them 👍 

|`city01`|
|---|
|![image](https://user-images.githubusercontent.com/20945684/28756818-31fd03d0-7576-11e7-84ba-6ef40ac7e05a.png)|
|Red rings outside the building|


|`city06`|
|---|
|![image](https://user-images.githubusercontent.com/20945684/28756730-54e9e72a-7574-11e7-9916-15fcd25d1dc2.png)|
|The right red ring is outside the building, the left one is on the street lamp, which isn't seleceted for the other structures.|

|`city17`|
|---|
|![image](https://user-images.githubusercontent.com/20945684/28756822-4369547a-7576-11e7-8d40-5e56885f54de.png)|
|Teal ring one is "outside the structure behind the building", I don't know if this should be covered|


|`city12` & `city13`|
|---|
|![image](https://user-images.githubusercontent.com/20945684/28756825-548a8026-7576-11e7-9d3a-2e7a29868bdc.png)|
|Red rings are outside|

reaperrr commented:
Updated, adressed last few nits.
